src/runtime/composables/local/useAuth.ts

```diff
@@ -25,43 +25,65 @@ const signIn: SignInFunc<Credentials, any> = async (credentials, signInOptions,
     params: signInParams ?? {}
   })
 
+  const { rawToken, rawRefreshToken } = useAuthState()
+
+  // Extract the access token
   const extractedToken = jsonPointerGet(response, config.token.signInResponseTokenPointer)
   if (typeof extractedToken !== 'string') {
-    console.error(`Auth: string token expected, received instead: ${JSON.stringify(extractedToken)}. Tried to find token at ${config.token.signInResponseTokenPointer} in ${JSON.stringify(response)}`)
+    console.error(
+      `Auth: string token expected, received instead: ${JSON.stringify(extractedToken)}.
+       Tried to find token at ${config.token.signInResponseTokenPointer} in ${JSON.stringify(response)}`
+    )
     return
   }
-
-  const { rawToken } = useAuthState()
   rawToken.value = extractedToken
 
+  // Extract the refresh token if enabled
+  if (config.refresh.isEnabled) {
+    const refreshTokenPointer = config.refresh.token.signInResponseRefreshTokenPointer
+
+    const extractedRefreshToken = jsonPointerGet(response, refreshTokenPointer)
+    if (typeof extractedRefreshToken !== 'string') {
+      console.error(
+        `Auth: string token expected, received instead: ${JSON.stringify(extractedRefreshToken)}.
+         Tried to find refresh token at ${refreshTokenPointer} in ${JSON.stringify(response)}`
+      )
+      return
+    }
+    rawRefreshToken.value = extractedRefreshToken
+  }
+
   await nextTick(getSession)
 
-  const { redirect = true } = signInOptions ?? {}
+  const { redirect = true, external } = signInOptions ?? {}
   let { callbackUrl } = signInOptions ?? {}
   if (typeof callbackUrl === 'undefined') {
     callbackUrl = await determineCallbackUrl(runtimeConfig.public.auth, () => getRequestURLWN(nuxt))
   }
   if (redirect) {
-    return navigateTo(callbackUrl)
+    return navigateTo(callbackUrl, { external })
   }
 }
 
 const signOut: SignOutFunc = async (signOutOptions) => {
   const nuxt = useNuxtApp()
   const runtimeConfig = await callWithNuxt(nuxt, useRuntimeConfig)
   const config = useTypedBackendConfig(runtimeConfig, 'local')
-  const { data, rawToken, token } = await callWithNuxt(nuxt, useAuthState)
+  const { data, token, rawToken, refreshToken, rawRefreshToken }: UseAuthStateReturn = await callWithNuxt(nuxt, useAuthState)
+
+  const signOutConfig = config.endpoints.signOut
 
   const headers = new Headers({ [config.token.headerName]: token.value } as HeadersInit)
+  const body = signOutConfig && signOutConfig.method.toLowerCase() === 'post' ? { refreshToken: refreshToken.value } : { }
```

Comment:
I don't understand where this comes from. I would advise you to not include extra changes to this PR, as it includes the surface area.

Instead, you can base a new PR on this branch and include the changes there so that we can review/test them independently of this refactor

Reply:
This functionality was added in https://github.com/sidebase/nuxt-auth/pull/731. I adjusted how the body is composed and also added a missing check if `refresh` is enabled.